pkg/api/handlers/utils/containers.go

```diff
@@ -202,12 +204,35 @@ func waitRemoved(ctrWait containerWaitFn) (int32, error) {
 	return code, err
 }
 
-func waitNextExit(ctrWait containerWaitFn) (int32, error) {
-	_, err := ctrWait(define.ContainerStateRunning)
-	if err != nil {
+func waitNextExit(ctx context.Context, containerName string) (int32, error) {
+	runtime := ctx.Value("runtime").(*libpod.Runtime)
+	var containerEngine entities.ContainerEngine = &abi.ContainerEngine{Libpod: runtime}
+	eventChannel := make(chan *events.Event)
+	errChannel := make(chan error)
+	opts := entities.EventsOptions{
+		EventChan: eventChannel,
+		Filter:    []string{"event=died", fmt.Sprintf("container=%s", containerName)},
+		Stream:    true,
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	go func() {
+		errChannel <- containerEngine.Events(ctx, opts)
+	}()
+	select {
+	case evt, ok := <-eventChannel:
+		if ok {
+			return int32(evt.ContainerExitCode), nil
+		}
+		select {
+		case err := <-errChannel:
+			return -1, err
+		default:
+			return -1, nil
```

Comment:
This should be an error, for instance, `internal error: unexpected empty data waiting for container`.

It should not happen but if it happens, we know where to look.

Reply:
I will rework this code.

Reply:
@vrothberg PTAL